src/dev/flang/be/interpreter/Intrinsics.java

```diff
@@ -911,12 +911,13 @@ static Callable effect(Interpreter interpreter, Clazz innerClazz)
         String in = innerClazz.feature().qualifiedName();   // == _fuir.clazzIntrinsicName(cl);
         switch (in)
           {
-          case "effect.replace": check(FuzionThread.current()._effects.get(cl) != null); FuzionThread.current()._effects.put(cl, m   );   break;
-          case "effect.default": if (FuzionThread.current()._effects.get(cl) == null) {  FuzionThread.current()._effects.put(cl, m   ); } break;
+          case "effect.replace": check(FuzionThread.current()._effects.get(cl.clazzOfType()) != null); FuzionThread.current()._effects.put(cl.clazzOfType(), m   );   break;
```

Comment:
Instead of calling `cl.clazzOfType()` repeatedly, can you initailize `var et = innerClazz._outer.effectType()` and then use `et` throughout instead of `cl`?